pkg/resources/task.go

```diff
@@ -598,7 +649,12 @@ func DeleteTask(ctx context.Context, d *schema.ResourceData, meta any) diag.Diag
 	tasksToResume, err := client.Tasks.SuspendRootTasks(ctx, id, id)
 	defer func() {
 		if err := client.Tasks.ResumeTasks(ctx, tasksToResume); err != nil {
-			log.Printf("[WARN] failed to resume tasks: %s", err)
+			log.Printf("[WARN] failed to resume tasks in delete: %s", err)
+			diags = append(diags, diag.Diagnostic{
```

Comment:
this diag is almost the same as the one above in create, right? (when deleting vs when creating) Let's extract this (and maybe even we don't need this distinction on when it happened - or you can parameterize it)

Reply:
I'll parametrize it; I still have to add it in the update method. After the first pr I see that some kind of error distinction is needed, especially in suspending/resuming that may happen multiple times during single resource apply.